apis/v1alpha2/validation/httproute.go

```diff
@@ -46,7 +46,38 @@ func ValidateHTTPRoute(route *gatewayv1a2.HTTPRoute) field.ErrorList {
 // validateHTTPRouteSpec validates that required fields of spec are set according to the
 // HTTPRoute specification.
 func validateHTTPRouteSpec(spec *gatewayv1a2.HTTPRouteSpec, path *field.Path) field.ErrorList {
-	return validateHTTPRouteUniqueFilters(spec.Rules, path.Child("rules"))
+	var errs field.ErrorList
+
+	errs = append(errs, validateHTTPRouteUniqueFilters(spec.Rules, path.Child("rules"))...)
+	errs = append(errs, validateHTTPRouteBackendServicePorts(spec.Rules, path.Child("rules"))...)
+
+	return errs
+}
+
+// validateHTTPRouteBackendServicePorts validates that v1.Service backends always have a port.
+func validateHTTPRouteBackendServicePorts(rules []gatewayv1a2.HTTPRouteRule, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+
+	for i, rule := range rules {
+		path = path.Index(i).Child("backendRefs")
+		for i, ref := range rule.BackendRefs {
+			if ref.BackendObjectReference.Group != nil &&
+				*ref.BackendObjectReference.Group != "" {
+				continue
+			}
+
+			if ref.BackendObjectReference.Kind != nil &&
```

Comment:
It seems that we don't have some documents to mark the default value of `Kind` is "Service".

Quote from the website:

> (Optional)
Kind is kind of the referent. For example “HTTPRoute” or “Service”.

Reply:
The default is set in the CRD:

https://github.com/kubernetes-sigs/gateway-api/blob/25607266669cc3616ed90404db4ff6e6d854a3b7/apis/v1alpha2/object_reference_types.go#L104-L106

Reply:
Yup, done.